tools/run_tests/run_xds_tests.py

```diff
@@ -2238,6 +2239,135 @@ def _delay(pct):
+                    'error-code-14,succeed-on-retry-attempt-4',
+                ),
+                (0, 0),  # all fail
+            ),
+            (
+                'retry-up-to-4-attempts',
+                (
+                    (4, "unavailable"),
+                    'error-code-14,succeed-on-retry-attempt-4',
+                ),
+                (num_rpcs, num_rpcs),  # all succeed
+            ),
+        ]
+
+        def _route(pri, name, retry_policy):
+            (num_retries, retry_on) = retry_policy
+            return {
+                'priority': pri,
+                'matchRules': [{
+                    'prefixMatch':
+                        '/',
+                    'headerMatches': [{
+                        'headerName': testcase_header,
+                        'exactMatch': name,
+                    }],
+                }],
+                'service': original_backend_service.url,
+                'routeAction': {
+                    'retryPolicy': {
+                        'retryConditions': [retry_on],
+                        'numRetries': num_retries,
+                    },
+                },
+            }
+
+        route_rules = []
+        pri = 0
+        for (testcase_name, test_config,
+             expected_succeeded_rpcs_range) in test_cases:
+            (retry_policy, rpc_behavior) = test_config
+            route_rules.append(_route(pri, testcase_name, retry_policy))
+            pri += 1
+
+        set_validate_for_proxyless(gcp, False)
+        patch_url_map_backend_service(gcp,
+                                      original_backend_service,
+                                      route_rules=route_rules)
+
+        first_case = True
+        for (testcase_name, test_config,
+             expected_succeeded_rpcs_range) in test_cases:
+            (retry_policy, rpc_behavior) = test_config
+            logger.info('starting case %s', testcase_name)
+
+            client_config = {}
+            client_config['metadata'] = [
+                (messages_pb2.ClientConfigureRequest.RpcType.UNARY_CALL,
+                 testcase_header, testcase_name),
+                (messages_pb2.ClientConfigureRequest.RpcType.UNARY_CALL,
+                 'rpc-behavior', rpc_behavior),
+            ]
+            client_config['rpc_types'] = [
+                messages_pb2.ClientConfigureRequest.RpcType.UNARY_CALL,
+            ]
+            configure_client(**client_config)
+            # wait a second to help ensure the client stops sending RPCs with
+            # the old config.  We will make multiple attempts if it is failing,
+            # but this improves confidence that the test is valid if the
+            # previous client_config would lead to the same results.
+            time.sleep(1)
+            # Each attempt takes 10 seconds; 20 attempts is equivalent to 200
+            # second timeout.
+            attempt_count = 20
+            if first_case:
+                attempt_count = 120
```

Comment:
Let's use attempt_count = 60. TD team promised configuration propagation should finish within 600s.

Reply:
Done.